src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
-
+        resize();
+        int hash = Math.abs(Objects.hashCode(key));
+        int index = hash % table.length;
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, hash, null);
+            ++size;
+        } else {
+            putInFullBucket(index, hash, key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> value;
+        return (value = getNode(key)) == null ? null : value.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putInFullBucket(int index, int hash, K key, V value) {
+        Node<K, V> currentNode = table[index];
+        if ((key == null && currentNode.key == null)
+                || (key != null && key.equals(currentNode.key))) {
+            currentNode.value = value;
+        } else {
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+                if ((key == null && currentNode.key == null)
+                        || (key != null && key.equals(currentNode.key))) {
+                    currentNode.value = value;
+                    return;
+                }
+            }
+            currentNode.next = new Node<>(key, value, hash, null);
+            ++size;
+        }
+    }
+
+    private void resize() {
+        if (size == threshold) {
+            size = 0;
+            Node<K, V>[] oldTable = table;
+            int oldCapacity = oldTable == null ? 0 : oldTable.length;
+            threshold = threshold << 1;
+            table = new Node[oldCapacity << 1];
+            if (oldTable != null) {
+                for (Node<K, V> node : oldTable) {
+                    if (node != null) {
+                        put(node.key, node.value);
+                        while (node.next != null) {
+                            node = node.next;
+                            put(node.key, node.value);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private Node<K, V> getNode(Object key) {
+        if (table.length > 0) {
+            for (Node<K, V> node : table) {
+                if (node != null) {
+                    if ((key == null && node.key == null)
+                            || (key != null && key.equals(node.key))) {
+                        return node;
+                    }
+                    while (node.next != null) {
+                        node = node.next;
+                        if ((key == null && node.key == null)
+                                || (key != null && key.equals(node.key))) {
+                            return node;
+                        }
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+        final int hash;
```

Comment:
I consider this as a redundant modification since the whole class is private and there is no access to this class except from inside of MyHashMap class.